File: src/types/image.ts
export interface MovieCover {
  id: number;
  code: string;
  image_path: string;
  is_primary: boolean;
  source: string;
}

export interface MovieScreenshot {
  id: number;
  code: string;
  image_path: string;
  sort_order: number;
}

export interface ActorImage {
  id: number;
  actor_id: number;
  image_path: string;
  sort_order: number;
}

File: src-tauri/src/models/image.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct MovieCover {
    pub id: i64,
    pub code: String,
    pub image_path: String,
    pub is_primary: bool,
    pub source: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct MovieScreenshot {
    pub id: i64,
    pub code: String,
    pub image_path: String,
    pub sort_order: i32,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ActorImage {
    pub id: i64,
    pub actor_id: i64,
    pub image_path: String,
    pub sort_order: i32,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SavedImage {
    pub id: i64,
    pub image_path: String,
}
